File: frontend/src/stores/broadcast.js
import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '@/services/api'

export const useBroadcastStore = defineStore('broadcast', () => {
  const broadcasts = ref([])
  const loading = ref(false)
  const error = ref(null)
  const selectedChats = ref([])
  const broadcastType = ref('favorites') // 'favorites' or 'forward'
  const message = ref('')

  async function sendBroadcast() {
    loading.value = true
    error.value = null
    try {
      const response = await api.post('/broadcast/send', {
        message: message.value,
        chat_ids: selectedChats.value,
        broadcast_type: broadcastType.value,
      })
      broadcasts.value.push(response.data)
      message.value = ''
      selectedChats.value = []
      return response.data
    } catch (err) {
      error.value = err.response?.data?.detail || 'Failed to send broadcast'
      throw error.value
    } finally {
      loading.value = false
    }
  }

  async function getHistory() {
    loading.value = true
    try {
      const response = await api.get('/broadcast/history')
      broadcasts.value = response.data.broadcasts
    } catch (err) {
      error.value = err.response?.data?.detail || 'Failed to fetch history'
    } finally {
      loading.value = false
    }
  }

  return {
    broadcasts,
    loading,
    error,
    selectedChats,
    broadcastType,
    message,
    sendBroadcast,
    getHistory,
  }
})

File: backend/app/api/routes/broadcast.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List

router = APIRouter()

class BroadcastMessage(BaseModel):
    message: str
    chat_ids: List[int]
    broadcast_type: str  # "favorites" or "forward"
    forward_from_chat: int = None

class BroadcastResponse(BaseModel):
    status: str
    sent_count: int
    failed_count: int

@router.post("/send", response_model=BroadcastResponse)
async def send_broadcast(data: BroadcastMessage):
    """Send broadcast message"""
    try:
        # TODO: Implement actual broadcast logic
        return BroadcastResponse(
            status="success",
            sent_count=len(data.chat_ids),
            failed_count=0
        )
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.get("/history")
async def get_broadcast_history():
    """Get broadcast history"""
    # TODO: Implement history retrieval
    return {"broadcasts": []}

@router.get("/stats")
async def get_broadcast_stats():
    """Get broadcast statistics"""
    # TODO: Implement stats
    return {
        "total_broadcasts": 0,
        "total_messages_sent": 0,
        "total_failed": 0
    }